src/lavinmq/http/controller.cr

```diff
@@ -102,11 +102,15 @@ module LavinMQ
         size = 0
         total = 0
         json.array do
-          iterator.each_with_index do |o, i|
-            total += 1
-            next if i < start || start + page_size <= i
-            o.to_json(json)
-            size += 1
+          begin
+            iterator.each_with_index do |o, i|
+              total += 1
+              next if i < start || start + page_size <= i
+              o.to_json(json)
+              size += 1
+            end
+          rescue e
+            json.object { json.field "error", e.message }
```

Comment:
Should we use the `{ error: type, reason: message }` format we use in the `ApiErrorHandler` ?
And set a different status code? 500?

Reply:
Yes, great!

Reply:
i'm not sure about the response code actually... if the result set is very large, and say 900 items have already been written to the response, then a) is it a valid or invalid response? b) the response header with the status code might already have been written to the socket and can't be changed

Reply:
True, like we fail to generate a response for one of 900 queues, this happens in rabbitmq as well, they just send some null fields back.

Reply:
So maybe we should just filter out the failure from items, and still render the queues with valid responses? (with a warning that we failed to load some data) ?